tools/src/main/python/dlpx/virtualization/_internal/commands/build.py

```diff
@@ -214,8 +214,23 @@ def prepare_upload_artifact(plugin_config_content, src_dir, schemas, manifest):
     if plugin_config_content.get('externalVersion'):
         artifact['externalVersion'] = plugin_config_content['externalVersion']
 
-    if plugin_config_content.get('luaName'):
+    if plugin_config_content.get(
```

Comment:
Can this validation be moved to plugin_validator.py? In plugin_validator.py, there is code like this - 
        self.__pre_import_checks = [
            self.__validate_plugin_config_content,
            self.__validate_plugin_entry_point,
            self.__check_for_undefined_names
        ]

So you can add one more method here "self.__check_for_lua_name_and_version" and in that method, all of this new code can go

Reply:
Forgot to mention - once you have the new method with this validation code, you will have to update this list of pre_import_checks and that way, your new method will get called when self.__run_checks() is called.

Reply:
I agree that this is much cleaner. Thanks for the suggestion!